src/instanceMethods/_main.js

```diff
@@ -120,7 +120,7 @@ export function _main (userParams) {
       switch (e.type) {
         case 'click':
           // Clicked 'confirm'
-          if (targetedConfirm && constructor.isVisible()) {
```

Comment:
I'm just reluctant in removing this check. Maybe I'm too scrupulous but there may be race conditions where the click event come when the constructor is _available_ but not visible. 

I tried to simulate a race condition in branch https://github.com/sweetalert2/sweetalert2/tree/fix/remove-excessive-isVisible-check-race-condition. I know that this example is quite stupid and probably doesn't represent a real use case but it shows the kind of unexpected behaviors I'm worried about. 

Having said that, probably the benefit of this change outweighs the risks, and I haven't followed the discussion on jest to closely to be able to judge the benefit. So feel free to ignore the comment...

Reply:
Am I doing something wrong or `remove-excessive-isVisible-check` and `remove-excessive-isVisible-check-race-condition` branches are identical?

Reply:
Alright, now I see that 

```js
Swal.close()
Swal.clickConfirm()
```

produces an error.

Will handle that case and cover it by test 👍

Reply:
Sorry, I forgot to commit my test (aaarg.. https://twitter.com/hashtag/NoDeploymentFridays?src=hash). It's on the branch now...

Reply:
@gverni I think this case should be handled by https://github.com/sweetalert2/sweetalert2/pull/1439/commits/9748d7b6e045adf4178983c6bdca412d843fcb99

Please review, thanks!